### PR TITLE
Primitives output schema

### PR DESCRIPTION
**Pull request by gadial**
# Summary

This PR adds schemas for output for V2 primitives (EstimatorV2/SamplerV2).

Addresses #[146](https://github.ibm.com/ibm-q-software/ws-primitives/issues/146)

### CODE REVIEW

File: schemas/sampler_result_v2_schema.json [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+                        "type": "object",
+                        "description": "PUB results for sampler are split according to the classical bits register names. For each register, an array of samples (in hexadecimal string format) is returned",
+                        "additionalProperties": {
+                            "type": "object",
+                            "description": "The data corresponding to a specific measurement",
+                            "properties": {
+                                "samples": {
+                                    "type": "array",
+                                    "description": "An array of samples (given as hexadecimal strings)",
+                                    "items": {"type":  "string"}
```
Comment (jyu00) [inline schemas/sampler_result_v2_schema.json:28]: When do we return a `samples` array? I thought sampled data is all grouped under register names?

File: tests/test_primitive_result_schemas.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            os.path.join(SCHEMAS_PATH, "sampler_result_v2_schema.json"), "r"
+        ) as fd:
+            self.sampler_schema = json.load(fd)
+        self.validator = jsonschema.Draft202012Validator(schema=self.sampler_schema)
+
+    def test_basic_result(self):
+        result_str = '{"results": [{"data": {"meas": {"samples": ["0x3", "0x3", "0x3", "0x3", "0x3"],' \
+                     ' "num_bits": 2}}, "metadata": {"circuit_metadata": {}}}], "metadata": {"version": 2}}'
+        result = json.loads(result_str)
+        print(result)
```
Comment (jyu00) [inline tests/test_primitive_result_schemas.py:63]: I don't think you meant to keep this

**Review by jyu00 — APPROVED** [review]
A couple of comments but overall looks good

File: schemas/sampler_result_v2_schema.json (hunk 1 above)
Comment (gadial) [reply schemas/sampler_result_v2_schema.json:28]: Yes, but currently the way `BitArray` is converted to JSON output is as an object with a `samples` field containing the flattening of the bit array and the number of bits.

File: tests/test_primitive_result_schemas.py (hunk 2 above)
Comment (gadial) [reply tests/test_primitive_result_schemas.py]: Removed in https://github.com/Qiskit/ibm-quantum-schemas/pull/15/commits/d8833921b23cd420d8dae9252cef3470180ea139

**Review by jyu00 — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**